BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managers/AgentRequestManager.java

```diff
@@ -225,6 +247,49 @@ public Response getRenderedConfigs(String serviceId) {
     return Response.ok(result).build();
   }
 
+  public Response purgeCache(String serviceId) {
+    LOG.info("purgeCache() called with serviceId={}", serviceId);
+
+    Optional<BaragonService> maybeService;
+    if (internalStateCache.containsKey(serviceId)){
+      maybeService = Optional.of(internalStateCache.get(serviceId).getService());
+    }
+    else {
+      maybeService = stateDatastore.getService(serviceId);
+    }
+
+    // 1. if no service exists, return an error response early
+    if (!maybeService.isPresent()){
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(String.format("Could not find service with serviceId=%s", serviceId)).build();
+    }
+
+    // 2. insert the serviceId hash into the purge URI
+    String purgeCacheUri = String.format(
+        baragonAgentConfiguration.getPurgeCacheUriFormat(),
+        Hashing.sha256()
+            .hashString(maybeService.get().getServiceId(), StandardCharsets.UTF_8)
+            .toString()
+    );
+    LOG.info("purgeCache() uri={}", purgeCacheUri);
+
+    // 3. now build the request
+    final HttpRequest.Builder builder = HttpRequest.newBuilder()
```

Comment:
what's our timeout on this request/client? We should make sure we can't accidentally hang in this. I don't expect it would happen, but would be quite messy to recover from if we did

Reply:
right now the request timeout is 10 seconds by default and the connection timeout is 5 seconds, this client is the one provided by `BaragonAgentServiceModule` which looks like it's already been tuned to avoid accidental hangs